homeassistant/components/sensibo/config_flow.py

```diff
@@ -28,7 +27,7 @@
 )
 
 
-async def async_validate_api(hass: HomeAssistant, api_key: str) -> bool:
+async def async_validate_api(hass: HomeAssistant, api_key: str) -> str:
```

Comment:
Making the caller have to check `if new_unique_id in [INVALID_AUTH, CANNOT_CONNECT, NO_DEVICES, NO_USERNAME]:` puts the burden on them to know how the validation works.

It would be cleaner to either raise a named exception or return a tuple here.

Reply:
Please move this into a `.util` or other named file since we usually don't import from the config flow file